Rarime/Code/Modules/ScanPassport/ViewModels/PassportViewModel.swift

```diff
@@ -107,57 +137,54 @@ class PassportViewModel: ObservableObject {
             
             LoggerUtil.common.info("Registering passport with circuit: \(registerIdentityCircuitName)")
             
-            var proof: ZkProof
-            if let registeredCircuitData = RegisteredCircuitData(rawValue: registerIdentityCircuitName) {
-                proof = try await generateGrothRegisterProof(
-                    user,
-                    passport,
-                    registerIdentityCircuitType,
-                    registeredCircuitData,
-                    &isCriticalRegistrationProcessInProgress
-                )
-            } else if let registeredCircuitData = RegisteredNoirCircuitData(rawValue: registerIdentityCircuitName) {
-                proof = try await generatePlonkRegisterProof(
-                    user,
-                    passport,
-                    registerIdentityCircuitType,
-                    registeredCircuitData,
-                    &isCriticalRegistrationProcessInProgress
-                )
-            } else {
+            let proof: ZkProof = try await {
+                if let data = RegisteredCircuitData(rawValue: registerIdentityCircuitName) {
+                    return try await generateGrothRegisterProof(
+                        user,
+                        passport,
+                        registerIdentityCircuitType,
+                        data,
+                        &isCriticalRegistrationProcessInProgress
+                    )
+                }
+                if let data = RegisteredNoirCircuitData(rawValue: registerIdentityCircuitName) {
+                    return try await generatePlonkRegisterProof(
+                        user,
+                        passport,
+                        registerIdentityCircuitType,
```

Comment:
why changed from this?
<img width="786" height="420" alt="image" src="https://github.com/user-attachments/assets/d2547ce2-53f0-44fd-b0a3-63138a510729" />